MinaProtocol/mina
# [develop] Reset feature flags before assertion + tests

- Use Proof_cache for Test_zkapp_update tests
- Port app_state test proof-cache to a JSON file
- Port delegate test proof-cache to a JSON file
- Port permissions test proof-cache to a JSON file
- Port token_symbol proof-cache to a JSON file
- Port verification_key test proof-cache to a JSON file
- Port voting_for test proof-cache to a JSON file
- Port zkapp_uri test proof-cache to a JSON file
- add alerts for stuck at catchup and bootstrap
- Introduce PipelineTag and PipelineFilter modules which can help more fluently describe which jobs should be run.
- hardcode
- pipeline
- Properly tag jobs
- comp
- move tag
- Revert "Auxiliary commit to revert individual files from 877400a0a0ed2bdcb76df77c74e1b2f83e67ba6c"
- format
- add timeout to invalid test
- add timeout to fuzzy zkapp tests
- add logs for compute transaction-snark
- change log level to info
- add alltests filter for backward compatibility
- Pickles: add runtime table test
- Bump up proof-systems to feature/blinding-for-table_id
- Chunking for public input
- Set the max_poly_size for the constraint system
- Propagate the useful error message from rust
- Remove unnecessary max_poly_size from fq_plonk_index
- Propagate step public input evals to the wrap proof
- Bump proof-systems for chunking bugfix
- Add num_chunks argument to Pickles.compile and friends
- Fix merges conflict
- Add zk_rows
- Use an array for public input evals
- Route zk_rows through to constant_term evaluation
- Update proof-systems for a zk_rows variable
- Rename variable to reflect zk_rows
- Bump proof-systems to appease clippy
- Rename variables
- Don't timeout if offline or bootstrapping before genesis
- factor common code
- Fix naming error
- Compute zk_polynomial and vanishes_on_zero_knowledge_and_previous_rows
- Verify chunked step proofs in the wrap circuit
- Bump proof-systems for clippy annoyances
- Bump proof-systems more; insert grumbling about clippy here
- Add docs for Processor, Load_data
- Add comment for version linter script
- Fixup Pickles.verify for chunked proofs
- Fix recursion over chunked proofs
- Bump proof-systems
- Add a test for chunking
- Update src/app/archive/lib/processor.ml
- Regenerate o1js bindings for zk_rows
- Update test for zk_rows
- Update cached vk for new zk_rows
- Reformat
- Re-update generated vk
- Add PROOF_CACHE_OUT env var, and reset proof_cache.json files
- Add an ERROR_ON_PROOF env var to fail when the cache is skipped
- Add README.md with caching info
- Update proof_caches for latest berkeley
- Add proof-cache for fee_payer and zkapp_deploy test
- Make zkapp_preconditions test deterministic, add proof-cache
- Make payments test deterministic, using a quickcheck generator
- Add proof-cache for zkapp_payments test
- Complicate code for review nitpick
- Commit link in readme
- Fix CI on non-default base branches
- Fix DirtyWhen attribute for snark profiler related changes
- fix comma
- Fix UNIQUE constraint for protocol versions
- Send txns more frequently in rosetta
- Dirty src/app/rosetta
- Increase relative bp stake
- Add crypto debugging helpers
- Absorb the runtime table commitment, when available
- Revert "Dirty src/app/rosetta"
- Add the runtime commitment to the IPA input
- Fix the proof-cache for the latest develop changes
- Propagate thread errors from o1trace even if they were raised during first cycle
- Reraise mina_net2 errors
- Empty commit to signpost Berkeley 2.0.0rampup6
- Reset feature flags before assertion
- Move Pickles custom gate circuits to their own library
- Hack around edge-case in proof-systems temporarily
- Add unit test for zkApps using optional custom gates
- Move test to zkapps_examples
